File: php/features.php
<?php
declare(strict_types=1);

// Nationalize SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class NationalizeFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new NationalizeBaseFeature();
            case "test":
                return new NationalizeTestFeature();
            default:
                return new NationalizeBaseFeature();
        }
    }
}

File: py/features.py
# Nationalize SDK feature factory

from feature.base_feature import NationalizeBaseFeature
from feature.test_feature import NationalizeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NationalizeBaseFeature(),
        "test": lambda: NationalizeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
